data/content_scripts/posting_process/sendkeys.js

```diff
@@ -0,0 +1,766 @@
+// Cross-broswer implementation of text ranges and selections
+// documentation: http://bililite.com/blog/2011/01/17/cross-browser-text-ranges-and-selections/
+// Version: 2.6
+// Copyright (c) 2013 Daniel Wachsstock
+// MIT license:
```

Comment:
More information now that I am not responding from my phone:

When you strip out functionality from a known library, we have the following issues:

* Mozilla's review process compares libraries against their source to ensure they are bit-exact. This is because it allows them to skip examining the library for add injectors, key loggers, etc. Including a modified library would likely lead to the extension being rejected.
* Whenever possible, we like to have code that is reusable across platforms because it makes it easier to port, test, and audit the code for multiple platforms.
* A common library would be more thoroughly tested than a modified version of the library, which means our modifications introduce potential bugs in addition to any issues we might experience integration our system with their library.

This is why I want you to checkout @SummerWish's proposal, but if that doesn't prove to be useful then you should just ignore me complaining about performance. Every line of code incurs a performance cost and it is better to invest time optimizing performance once we know what is the most costly. If the privly.js library is 1000x slower than the typing simulation code, then it is wasting effort to optimize the typing code before working on privly.js